projects/driver_controls/test/test_pedal_calibration_test.c

```diff
@@ -0,0 +1,139 @@
+    equal &= (result->zone_thresholds_main[zone].min == expected->zone_thresholds_main[zone].min) &&
+             (result->zone_thresholds_main[zone].max == expected->zone_thresholds_main[zone].max);
+    LOG_DEBUG("zone thresholds: min: %d == %d - max: %d == %d\n",
+              result->zone_thresholds_main[zone].min, expected->zone_thresholds_main[zone].min,
+              result->zone_thresholds_main[zone].max, expected->zone_thresholds_main[zone].max);
+  }
+  for (ThrottleChannel channel = THROTTLE_CHANNEL_MAIN; channel < NUM_THROTTLE_CHANNELS;
+       channel++) {
+    equal &=
+        (result->line[channel].full_brake_reading == expected->line[channel].full_brake_reading) &&
+        (result->line[channel].full_throttle_reading ==
+         expected->line[channel].full_throttle_reading);
+    LOG_DEBUG("line: brake: %d == %d - throttle: %d == %d\n",
+              result->line[channel].full_brake_reading, expected->line[channel].full_brake_reading,
+              result->line[channel].full_throttle_reading,
+              expected->line[channel].full_throttle_reading);
+  }
+  equal &= result->tolerance == expected->tolerance;
+  LOG_DEBUG("tolerance: %d == %d\n", result->tolerance, expected->tolerance);
+  equal &= result->channel_main == expected->channel_main;
+  LOG_DEBUG("channel main: %d == %d\n", result->channel_main, expected->channel_main);
+  equal &= result->channel_secondary == expected->channel_secondary;
+  LOG_DEBUG("channel secondary: %d == %d\n", result->channel_secondary,
+            expected->channel_secondary);
+  return equal;
+}
+
+void setup_test(void) {
+  gpio_init();
+  interrupt_init();
+  gpio_it_init();
+  soft_timer_init();
+  I2CSettings i2c_settings = {
+    .speed = I2C_SPEED_FAST,                    //
+    .scl = { .port = GPIO_PORT_B, .pin = 10 },  //
+    .sda = { .port = GPIO_PORT_B, .pin = 11 },  //
+  };
+  i2c_init(TEST_ADS1015_I2C_PORT, &i2c_settings);
+  GPIOAddress ready_pin = {
+    .port = GPIO_PORT_B,  //
+    .pin = 2,             //
+  };
+  ads1015_init(&s_ads1015_storage, TEST_ADS1015_I2C_PORT, TEST_ADS1015_ADDR, &ready_pin);
+  pedal_calibration_init(
+      &s_storage, &s_ads1015_storage, TEST_PEDAL_CALIBRATION_ADC_CHANNEL_A,
+      TEST_PEDAL_CALIBRATION_ADC_CHANNEL_B, TEST_PEDAL_CALIBRATION_BRAKE_PERCENTAGE,
+      TEST_PEDAL_CALIBRATION_COAST_PERCENTAGE, TEST_PEDAL_CALIBRATION_SAFETY_FACTOR);
+}
+
+void test_pedal_calibration_correctness(void) {
+  memset(s_sample_counter, 0, sizeof(s_sample_counter));
+  s_current_state = PEDAL_CALIBRATION_STATE_FULL_BRAKE;
+  pedal_calibration_process_state(&s_storage, PEDAL_CALIBRATION_STATE_FULL_BRAKE);
+
+  memset(s_sample_counter, 0, sizeof(s_sample_counter));
+  s_current_state = PEDAL_CALIBRATION_STATE_FULL_THROTTLE;
+  pedal_calibration_process_state(&s_storage, PEDAL_CALIBRATION_STATE_FULL_THROTTLE);
+
+  pedal_calibration_calculate(&s_storage, &s_throttle_calibration_data);
+  ThrottleCalibrationData expected = {
```

Comment:
You probably shouldn't be hardcoding your math in the test. It would probably be more useful to run the throttle module with your calibration data.